cfme/configure/access_control.py

```diff
@@ -850,43 +1061,151 @@ def create(self, cancel=False):
             view.add_button.click()
             flash_message = 'Role "{}" was saved'.format(self.name)
         view = self.create_view(AllRolesView)
+
+        try:
+            view.flash.assert_message(flash_blocked_msg)
+            raise RBACOperationBlocked(flash_blocked_msg)
+        except AssertionError:
+            pass
+
         view.flash.assert_success_message(flash_message)
+
         assert view.is_displayed
 
-    def update(self, updates):
+    def _update_using_all_selection(self):
+        """
+        Update existing role by selecting it from the access control role list
+
+        Throws:
+            RBACOperationBlocked: If operation is blocked due to current user
+                not having appropriate permissions OR update is not allowed
+                for currently selected role
+        """
+        flash_blocked_msg = 'Read Only Role "{}" can not be edited'.format(self.name)
+        name_column = "Name"
+        find_row_kwargs = {name_column: self.name}
+
+        view = navigate_to(Role, 'All')
+        try:
+            role_row = view.paginator.find_row_on_pages(view.table, **find_row_kwargs)
+            # Zero index is the unnamed column with the checkbox
+            role_row[0].check()
+        except NoSuchElementException:
+            raise Exception("Failed to find role row with name {}".format(self.name))
+
+        view.configuration.item_select('Edit the selected Role')
+
+        try:
+            view.flash.assert_message(flash_blocked_msg)
+            raise RBACOperationBlocked(flash_blocked_msg)
+        except AssertionError:
+            pass
+
+    def update(self, updates, all_selection=False):
         """ Update role method
             Args:
                 updates: role data that should be changed
         Note: In case updates is the same as original role data, update will be canceled,
         as 'Save' button will not be active
         """
-        view = navigate_to(self, 'Edit')
-        changed = view.fill({
-            'name_txt': updates.get('name'),
-            'vm_restriction_select': updates.get('vm_restriction')
-        })
-        feature_changed = self.set_role_product_features(view, updates.get('product_features'))
-        if changed or feature_changed:
-            view.save_button.click()
-            flash_message = 'Role "{}" was saved'.format(updates.get('name', self.name))
+        flash_blocked_msg = VersionPick({
+            "5.7": ("Read Only Role \"{}\" can not be edited")}).pick(
+                self.appliance.version).format(self.name)
```

Comment:
It's removed it. It was a holdover from a 5.6 test string. I left the version pick in as a quick way to easily insert a string version update.